src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AppointmentServiceTest.kt

```diff
@@ -7,87 +7,117 @@ import com.nhaarman.mockitokotlin2.times
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Attended
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Attended.NO
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.AppointmentRepository
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.AuthUserRepository
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.util.AppointmentFactory
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.util.AuthUserFactory
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.util.ReferralFactory
 import java.time.OffsetDateTime
+import java.util.UUID
 
 class AppointmentServiceTest {
   private val authUserRepository: AuthUserRepository = mock()
   private val appointmentRepository: AppointmentRepository = mock()
+  private val communityAPIBookingService: CommunityAPIBookingService = mock()
 
   private val authUserFactory = AuthUserFactory()
   private val appointmentFactory = AppointmentFactory()
+  private val referralFactory = ReferralFactory()
+
+  private val createdByUser = authUserFactory.create()
 
   private val appointmentService = AppointmentService(
     appointmentRepository,
+    communityAPIBookingService,
     authUserRepository,
   )
 
+  @BeforeEach
+  fun beforeEach() {
+    whenever(authUserRepository.save(any())).thenReturn(createdByUser)
+  }
+
   @Test
   fun `create new appointment if none currently exist`() {
+    // Given
     val durationInMinutes = 60
     val appointmentTime = OffsetDateTime.parse("2020-12-04T10:42:43+00:00")
-    val createdByUser = authUserFactory.create()
-    val appointment = appointmentFactory.create()
-
-    whenever(appointmentRepository.save(any())).thenReturn(appointment)
-    whenever(authUserRepository.save(any())).thenReturn(createdByUser)
-
-    appointmentService.createOrUpdateAppointment(null, durationInMinutes, appointmentTime, createdByUser)
-
-    val argumentCaptor = argumentCaptor<Appointment>()
-    verify(appointmentRepository, times(1)).save(argumentCaptor.capture())
-    val arguments = argumentCaptor.firstValue
-
-    assertThat(arguments.appointmentTime).isEqualTo(appointmentTime)
-    assertThat(arguments.durationInMinutes).isEqualTo(durationInMinutes)
+    val referral = referralFactory.createSent()
+    val deliusAppointmentId = 99L
+
+    whenever(communityAPIBookingService.book(referral, null, appointmentTime, durationInMinutes, SUPPLIER_ASSESSMENT))
+      .thenReturn(deliusAppointmentId)
+    val savedAppointment = appointmentFactory.create(
+      appointmentTime = appointmentTime,
+      durationInMinutes = durationInMinutes,
+      deliusAppointmentId = deliusAppointmentId,
+    )
+    whenever(appointmentRepository.save(any())).thenReturn(savedAppointment)
+
+    // When
+    val newAppointment = appointmentService.createOrUpdateAppointment(referral, null, durationInMinutes, appointmentTime, SUPPLIER_ASSESSMENT, createdByUser)
+
+    // Then
```

Comment:
Would it be good to have a conversation to pick a consistent style throughout the project?